enhancements/monitoring/multi-tenant-alerting.md

```diff
@@ -0,0 +1,601 @@
+receivers:
+# The next item is generated from an AlertmanagerConfig resource named alertmanagerconfig1 in namespace foo.
+- name: foo-alertmanagerconfig1-my-receiver
+  ...
+```
+
+#### RBAC
+
+The cluster monitoring operator ships a new cluster role
+`alertmanager-config-edit` that grants all actions on `AlertmanagerConfig`
+custom resources.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: alertmanager-config-edit
+rules:
+- apiGroups:
+  - monitoring.coreos.com
+  resources:
+  - alertmanagerconfigs
+  verbs:
+  - '*'
+```
+
+Cluster admins can bind the cluster role with a `RoleBinding` to grant
+permissions to users or groups on `AlertmanagerConfig` custom resources within
+a given namespace.
+
+```bash
+oc -n <namespace> adm policy add-role-to-user \
+  alertmanager-config-edit <user> --role-namespace <namespace>
+```
+
+This role complements the existing `monitoring-edit`, `monitoring-rules-edit` and `monitoring-rules-view` roles.
+
+#### Resource impacts
+
+The size of the Alertmanager routing tree can grow to thousands of entries
+(possibly several `AlertmanagerConfig` resources per namespace) and this may
+hinder the performances of the Plaform Alertmanager (increased latency of alert
+notifications for instance).
+
+We already know from upstream users that Alertmanager can deal with many
+routes. We plan to simulate environments with thousands of `AlertmanagerConfig`
+resources and measure the impact on notification delivery.
+
+### Risks and Mitigations
+
+#### Disruption of Alertmanager
+
+Even though the Prometheus operator prevents it as much as it can, it may be
+possible for users to create an `AlertmanagerConfig` resource that triggers the
+Prometheus operator to generate an invalid Alertmanager configuration, leading
+to a potential outage of the Alertmanager cluster.
+
+Mitigations
+* If cluster admins have configured an external notification provider coupled with the always firing `Watchdog` alert, they should receive an out-of-band notification about the alerting pipeline being broken.
+* The `AlertmanagerBadConfig` alert fires when Alertmanager can't reload its configuration.
+* Cluster admins can exclude specific user namespaces (once the "rogue" `AlertmanagerConfig` resource(s) have been identified) to restore UWM functionality for good citizens.
+* When alerts are sent to the Platform Alertmanager, cluster admins can turn off the support for `AlertmanagerConfig` in the CMO configmap so that the Platform Alertmanager cluster can process platform alerts again and the cluster admins have time to identiy the "rogue" `AlertmanagerConfig` resource(s).
+
+#### Misconfiguration of receivers
+
+Users may provide bad credentials for the receivers, the system receiving the
+notifications might be unreachable, or the system might be unable to process
+the requests. These situations would trigger the
+`AlertmanagerFailedToSendAlerts` and/or `AlertmanagerClusterFailedToSendAlerts`
```

Comment:
In the context of external AlertManager and/or UWM/AlertManager - these alerts routed via openshift-monitoring/alertmanger thus alerting SRE that a customer has miss-configured their own AlertManager stack ? That is ok, as we can short circuit this use-case to service log and remove SRE from the loop... just wanted to clarify.

Reply:
this is correct: these alerts will be routed to the Platform Alertmanager.